### PR TITLE
53 doors

### LINKED ISSUE #53: doors

### PR DESCRIPTION
**Pull request by antoineverin**
- add D to valid chars in map
- add type in hit struct
- update raycast to take other type of wall and store his type
- update print_img to take wall type as parameter for texture
- add wall_state struct
- add wall_state var to graphic struct
- move vector delcaration to graphic
- refactor parse_line
- add parse_door function
- add wall_state struct
- refactor raycast
- rename wall_state struct to ws
- add door functions
- add option in raycast
- fix get_wall_state not going through array
- add position to hit
- add close_door function
- update check_collision to take doors
- add action close_door to e
- refactor move collision check and add support for door collisions
- norminetted check_collision in raycast
- norminetted graphic.h
- move create_rectangle to parse_map_utils

### CODE REVIEW

File: include/error.h
Comment (Neffi42) [inline include/error.h:1]: useless

File: src/loop.c [hunk 1]
```diff
@@ -40,6 +40,8 @@ static int	handle_key(int keycode, t_data *data)
 		forward(STEP, &data->player, data->map);
 	if (keycode == XK_s)
 		forward(-STEP, &data->player, data->map);
+	if (keycode == XK_e)
```
Comment (Neffi42) [inline src/loop.c:43]: how do you open the door ?

File: src/parse_map.c [hunk 2]
```diff
@@ -3,42 +3,16 @@
...
-		}
-		len = -1;
-		while (((char *)line->content)[++len])
-			if (((char *)line->content)[len] == ' ')
-				((char *)line->content)[len] = '\0';
-		map->content[i++] = line->content;
-		line = line->next;
-	}
-}
+#define DOOR "D"
```
Comment (Neffi42) [inline src/parse_map.c]: why not in header file ?

**Review by Neffi42 — CHANGES_REQUESTED** [review]

File: src/loop.c (hunk 1 above)
Comment (iamantoinev) [reply src/loop.c:43]: by pressing `e` from at least 1 of distance

File: include/error.h
Comment (iamantoinev) [reply include/error.h:1]: was modified before `VALID_CHARS` moved

**Review by Neffi42 — APPROVED** [review]

**Pull request merged**